batch/src/main/java/org/openmrs/analytics/JdbcFetchUtil.java

```diff
@@ -171,11 +172,17 @@ public void processElement(@Element KV<String, Iterable<String>> element,
 		return pipeline.apply(Create.of(uuids).withCoder(pipeline.getCoderRegistry().getCoder(String.class)));
 	}
 	
-	public PCollection<String> fetchAllUuids(Pipeline pipeline, String tableName, int jdbcFetchSize) throws SQLException {
+	public PCollection<String> fetchAllUuids(Pipeline pipeline, String tableName, int jdbcFetchSize)
+	        throws SQLException, CannotProvideCoderException {
 		int maxId = fetchMaxId(tableName);
 		log.info(String.format("Will fetch up to %d rows from table %s", maxId, tableName));
 		Map<Integer, Integer> idRanges = createIdRanges(maxId, jdbcFetchSize);
-		return pipeline.apply(Create.of(idRanges)).apply(new JdbcFetchUtil.FetchUuids(tableName, getJdbcConfig()));
+		// We need to specify the coder in case idRanges are empty.
+		return pipeline
```

Comment:
I think it is actually better to return an empty PCollection in this case because you can check if `idRanges` is empty at the time of pipeline construction. I would suggest to even move the check before `createIdRanges` because I think in this case you can make that decision from `maxId` directly.